Tests/js/views/dashboard/assets/ez-dashboardblockallcontentview-tests.js

```diff
@@ -0,0 +1,245 @@
+            this.view.destroy();
+        },
+
+        'Should render view correctly with loading state': function () {
+            var view = this.view,
+                templateCalled = false,
+                origTpl = view.template,
+                container = view.get('container');
+
+            view.template = function () {
+                templateCalled = true;
+
+                return origTpl.apply(this, arguments);
+            };
+
+            view.render();
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+            Y.Assert.areSame(0, container.one(SELECTOR_CONTENT).all('tr').size(), 'Should not render any row');
+            Y.Assert.isTrue(container.hasClass(CLASS_LOADING), 'Should add the loading state CSS class to the view container');
+        },
+
+        'Should render view correctly with content inside': function () {
+            var view = this.view,
+                contentMock = new Y.Mock(),
+                versionMock = new Y.Mock(),
+                contentTypeMock = new Y.Mock(),
+                modificationDate = new Date().toString(),
+                templateCalled = false,
+                origTpl = view.template,
+                container = view.get('container'),
+                prepareMock = function (mock, values, callCount) {
+                    Y.Mock.expect(mock, {
+                        method: 'get',
+                        args: [Y.Mock.Value.String],
+                        run: function (key) {
+                            if (values[key]) {
+                                return values[key];
+                            } else {
+                                Y.Assert.fail('Trying to get unexpected attribute value: ' + key);
+                            }
+                        },
+                        callCount: callCount
+                    });
+                };
+
+            prepareMock(contentMock, {
+                id: 'alter-bridge',
+                name: 'Alter Bridge',
+                resources: {MainLocation: '/location/1/2/3'},
+                mainLanguageCode: 'eng-GB',
+                currentVersion: versionMock
+            }, 5);
+
+            prepareMock(versionMock, {
+                versionNo: '2013',
+                modificationDate: modificationDate
+            }, 2);
+
+            prepareMock(contentTypeMock, {
+                names: {'eng-GB': 'Rock'}
+            }, 1);
+
+            view.template = function () {
+                templateCalled = true;
+
+                return origTpl.apply(this, arguments);
+            };
+
+            view.set('contentList', [{
+                content: contentMock,
+                contentType: contentTypeMock
+            }]);
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+            Y.Assert.areSame(1, container.one(SELECTOR_CONTENT).all('tr').size(), 'Should render 1 row');
```

Comment:
by asserting that you're not testing the view but its template. Instead, you should test the variable available in the template to make it's consistent with the `contentList` you set. The actual output of the view will be tested in Behat tests.